src/mixins/document-mixin.ts

```diff
@@ -63,12 +63,16 @@ export default class DocumentMixin extends Vue {
     const pdfBufferData = await file.arrayBuffer()
     const pdfData = new Uint8Array(pdfBufferData) // put it in a Uint8Array
     const pdf = await pdfjsLib.getDocument({ data: pdfData })
-    const p1 = await pdf.getPage(1)
-    const [x, y, w, h] = p1._pageInfo.view
-    const width = w - x
-    const height = h - y
-    const isvalidPageSize = (width / pageSizeInfo.pointsPerInch === pageSizeInfo.width) &&
-      (height / pageSizeInfo.pointsPerInch === pageSizeInfo.height)
+    let isvalidPageSize = true
+    for (let pageNum = 1; pageNum <= pdf.numPages; pageNum++) {
+      const p1 = await pdf.getPage(pageNum)
+      const [x, y, w, h] = p1._pageInfo.view
+      const width = w - x
+      const height = h - y
+      isvalidPageSize = (width / pageSizeInfo.pointsPerInch === pageSizeInfo.width) &&
+        (height / pageSizeInfo.pointsPerInch === pageSizeInfo.height)
+      if (!isvalidPageSize) return isvalidPageSize
```

Comment:
You might as well just return False here, since you've found an invalid page.

Also, you might as well return True on line 76.

Reply:
Personally, I like coding things as false by default, then whatever's being tested must prove itself as true.

Reply:
That works, too.

A couple of things I look for:
- Easy to understand. In this code, I had to look around and run the code in my head to verify what values are being returned.
- I like short-circuiting results (which you've done here) 👍 so I'm ok with returning False when you've found an exception.

Reply:
It's done. It explicitly returns false in the loop.